src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -752,7 +752,6 @@ internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNo
         {
             Debug.Assert(expr is object);
             Debug.Assert(expr.Type is object);
-            // PROTOTYPE(RefStructInterfaces): adjust?
             Debug.Assert(expr.Type.IsRefLikeType || hasAwait); // pattern dispose lookup is only valid on ref structs or asynchronous usings
```

Comment:
Is pattern dispose not allowed on `allows ref struct` type parameters? Do we have tests for that?

Reply:
> Is pattern dispose not allowed on allows ref struct type parameters? Do we have tests for that?

I think `Using_06` covers that. The fact is also explicitly stated in [https://github.com/dotnet/csharplang/blob/main/proposals/ref-struct-interfaces.md#using-statement](https://github.com/dotnet/csharplang/blob/main/proposals/ref-struct-interfaces.md#using-statement) section.